src/libs/Navigation/AppNavigator/AuthScreens.tsx

```diff
@@ -604,15 +606,20 @@ const AuthScreensMemoized = memo(AuthScreens, () => true);
 // Migration to useOnyx cause re-login if logout from deeplinked report in desktop app
 // Further analysis required and more details can be seen here:
 // https://github.com/Expensify/App/issues/50560
-// eslint-disable-next-line
-export default withOnyx<AuthScreensProps, AuthScreensProps>({
-    session: {
-        key: ONYXKEYS.SESSION,
-    },
-    lastOpenedPublicRoomID: {
-        key: ONYXKEYS.LAST_OPENED_PUBLIC_ROOM_ID,
-    },
-    initialLastUpdateIDAppliedToClient: {
-        key: ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT,
-    },
-})(AuthScreensMemoized);
+export default function AuthScreensWithOnyx() {
+    const [session, sessionStatus] = useOnyx(ONYXKEYS.SESSION);
+    const [lastOpenedPublicRoomID, lastOpenedPublicRoomIDStatus] = useOnyx(ONYXKEYS.LAST_OPENED_PUBLIC_ROOM_ID);
+    const [initialLastUpdateIDAppliedToClient, initialLastUpdateIDAppliedToClientStatus] = useOnyx(ONYXKEYS.ONYX_UPDATES_LAST_UPDATE_ID_APPLIED_TO_CLIENT);
+
+    if (isLoadingOnyxValue(sessionStatus, lastOpenedPublicRoomIDStatus, initialLastUpdateIDAppliedToClientStatus)) {
+        return <FullScreenLoadingIndicator />;
+    }
+
+    return (
+        <AuthScreensMemoized
```

Comment:
I see we usually fetch the data with `useOnyx` directly in the main function. This way of using another parent and passing the Onyx fetched values as props is new to me. Any benefit of doing this vis-a-vis fetching these values directly with `useOnyx` in the existing main `AuthScreens` function?

Reply:
I have followed this approach, @blazejkustra Has explained pretty neat here.
https://github.com/Expensify/App/issues/49103#issuecomment-2385325701